os/arch/arm/src/armv7-m/up_busfault.c

```diff
@@ -0,0 +1,148 @@
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <assert.h>
+#include <debug.h>
+
+#include <arch/irq.h>
+
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
+#include <tinyara/reboot_reason.h>
+#include <arch/reboot_reason.h>
+#endif
+
+#include "up_arch.h"
+#include "nvic.h"
+#include "up_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define BFARVALID	0x00008000
+#define LSPERR	0x00002000
+#define STKERR	0x00001000
+#define UNSTKERR	0x00000800
+#define IMPRECISERR	0x00000400
+#define PRECISERR	0x00000200
+#define IBUSERR	0x00000100
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_busfault
+ *
+ * Description:
+ *   This is Bus Fault exception handler.
+ *
+ ****************************************************************************/
+
+int up_busfault(int irq, FAR void *context, FAR void *arg)
+{
+	(void)irqsave();
+	uint32_t *regs = (uint32_t *)context;
+	uint32_t cfsr = getreg32(NVIC_CFAULTS);
+	uint32_t bfar = getreg32(NVIC_BFAULT_ADDR);
+	lldbg("PANIC!!! Bus fault occured while executing instruction at address : 0x%08x\n", regs[REG_R15]);
+	lldbg("CFAULTS: 0x%08x\n", cfsr);
+
+	if (cfsr & BFARVALID) {
+		lldbg("Fault occured while accessing address (BFAR) : 0x%08x\n", bfar);
+	} else {
+		lldbg("Unable to determine fault address.\n");
+	}
+
+	if (cfsr & PRECISERR) {
+		fltdbg("Precise data access error occured.\n");
+	} else if (cfsr & IMPRECISERR) {
+		fltdbg("Imprecise data access error occured.\n");
+	} else if (cfsr & STKERR) {
+		fltdbg("Error while stacking registers during exception entry.\n");
+	} else if (cfsr & UNSTKERR) {
+		fltdbg("Error while unstacking registers during exception return.\n");
+	} else if (cfsr & LSPERR) {
+		fltdbg("Error occured during lazy state preservation of Floating Point unit registers.\n");
+	} else if (cfsr & IBUSERR) {
+		fltdbg("Error on an instruction prefetch.\n");
+	}
```

Comment:
Is there any specific way to resolve each issue?
What I mean is
1. Ok. I can know what error is happened.
2. What should I check to resolve each issue?

Reply:
@sunghan-chang 
There is not simple method for resolving each issue. 
We are already printing the PC value which caused the fault. Then, we need to manually check the code at that location and see why the issue happened. 
I propose that we must merge this pr after addressing other comments. 
We will check if your suggestion is possible and raise a new PR for it.

Reply:
sure. thank you for reply.